Final Legal Chatbot Files/ollama_llm.py

```
@@ -0,0 +1,17 @@
+
+class OllamaLLM(LLM):
+    def __init__(self, model_name="deepseek-r1:1.5b"):
+        self.llm = Ollama(model=model_name)
+        self.prompt_template = FIFTH_GRADE_PROMPT_TEMPLATE + "\nHuman: {question}"
+
+    def generate_response(self, question: str, context: str) -> str:
+        prompt = self.prompt_template.format(context=context, question=question)
+        try:
+            response = self.llm(prompt)
+            return response
+        except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)